go/exocom/main_test.go

```diff
@@ -189,16 +189,29 @@ func FeatureContext(s *godog.Suite) {
 			return fmt.Errorf("Expected to receive a message but got %v", err)
 		}
 		outgoingMessageActivityID = actualMessage.ActivityID
-		payload, ok := actualMessage.Payload.(map[string]interface{})
-		if !ok {
-			return errors.New("Error: cannot parse payload")
+
+		expectedBytes, err := json.Marshal(expectedMessage.Payload)
+		if err != nil {
+			return fmt.Errorf(err.Error())
+		}
+		actualBytes, err := json.Marshal(actualMessage.Payload)
+		if err != nil {
+			return fmt.Errorf(err.Error())
+		}
+		actualMessagePayload := structs.Message{}
+		expectedMessagePayload := structs.Message{}
+		err = json.Unmarshal(actualBytes, &actualMessagePayload)
+		if err != nil {
+			return fmt.Errorf(err.Error())
+		}
+		err = json.Unmarshal(expectedBytes, &expectedMessagePayload)
+		if err != nil {
+			return fmt.Errorf(err.Error())
 		}
-		delete(payload, "sessionId")
-		delete(payload, "responseTime")
-		delete(payload, "sender")
-		delete(payload, "timestamp")
-		if !reflect.DeepEqual(expectedMessage.Payload, payload) {
-			return fmt.Errorf("Expected payload to equal %s but got %s", expectedMessage.Payload, actualMessage.Payload)
+		actualMessagePayload.ID = expectedMessagePayload.ID
+		actualMessagePayload.ActivityID = expectedMessagePayload.ActivityID
```

Comment:
Are these two line even needed? Based on the added tests, they don't appear to be as the id and activityId are not generated by exocom but passed in